### PR TITLE
Small refactoring of the JIT facade to align it with adapters.

### PR DESCRIPTION
**Pull request by ryzhyk**
This is part of work on #678

### CODE REVIEW

File: crates/dataflow-jit/src/codegen/json/deserialize.rs [hunk 1]
```diff
@@ -51,8 +52,8 @@ pub unsafe fn call_deserialize_fn(
     } else {
         #[inline(never)]
         #[cold]
-        fn deserialize_error(error: String) -> Box<dyn std::error::Error> {
-            format!("an error occurred while parsing the key {error}").into()
+        fn deserialize_error(error: String) -> AnyError {
+            anyhow!(format!("an error occurred while parsing the key {error}"))
```
Comment (Kixiron) [inline crates/dataflow-jit/src/codegen/json/deserialize.rs:56]:
```suggestion
            anyhow!("an error occurred while parsing the key {error}")
```

File: crates/dataflow-jit/src/facade/handle.rs [hunk 2]
```diff
@@ -5,18 +5,56 @@ use crate::{
...
+    /// Buffer a new update.
+    ///
+    /// Returns an error if deserialization fails, i.e., the serialized
+    /// representation is corrupted or does not match the value type of
+    /// the underlying input stream.
+    fn push(&mut self, data: &[u8], weight: i32) -> AnyResult<()>;
+
+    /// Reserve space for at least `reservation` more updates in the
+    /// internal input buffer.
+    fn reserve(&mut self, reservation: usize);
```
Comment (Kixiron) [inline crates/dataflow-jit/src/facade/handle.rs:28]:
```suggestion
    fn reserve(&mut self, additional: usize);
```

File: crates/dataflow-jit/src/facade/handle.rs [hunk 3]
```diff
@@ -26,23 +64,42 @@ impl JsonSetHandle {
             handle,
             deserialize_fn,
             vtable,
+            updates: Vec::new(),
+        }
+    }
+
+    fn clear(&mut self) {
+        if self.updates.capacity() > MAX_REUSABLE_CAPACITY {
+            self.updates = Vec::new();
```
Comment (Kixiron) [inline crates/dataflow-jit/src/facade/handle.rs:73]: I think `self.updates.shrink_to(MAX_REUSABLE_CAPACITY)` might be better

**Review by Kixiron — APPROVED** [review]

File: crates/dataflow-jit/src/codegen/json/deserialize.rs (hunk 1 above)
Comment (ryzhyk) [reply crates/dataflow-jit/src/codegen/json/deserialize.rs:56]: In fact, I should just be outputting `error`, it's up to the caller to build an error message with it.

**Pull request merged**